requests/adapters.py

```diff
@@ -493,13 +493,12 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                 # TODO: Remove this in 3.0.0: see #2811
                 if not isinstance(e.reason, NewConnectionError):
                     raise ConnectTimeout(e, request=request)
-
-            if isinstance(e.reason, ResponseError):
+            elif isinstance(e.reason, ResponseError):
```

Comment:
@bobstrecansky while I understand that was your intent and that this was an attempt to fix that, your attempt has done little, if anything, to actually fix that. I think you're missing a critical part of the problem here:

Requests lives at the top-level of several layers of libraries. Immediately below Requests is urllib3. *Most* of the exceptions we handle and then *re-wrap* are urllib3 exceptions. Some of those exceptions are simple exceptions raised by urllib3. Others are exceptions raised by *other* libraries that urllib3 relies on and are re-wrapped. For example, sometimes urllib3 may raise an `SSLError` that is wrapping an exception from the standard library's `ssl` module. Othertimes the `SSLError` may be wrapping an exception from `OpenSSL`. It depends on the user's installation. It's also plausible for urllib3 to be wrapping an exception from `httplib` or `http.client` (depending on Python 2 or Python 3). Beyond that each of those libraries may be wrapping some other library's exception.

The confusing part here stems from two things:

1. Errors wrapped in errors wrapped in errors wrapped in ....
2. Overly generic exceptions being raised in Requests or urllib3 that while technically accurate do not immediately portray the whole story.